File: rust/kernel-python/test-themes.py
#!/usr/bin/env python3

"""
Test script for visual verification of Stencila theme variables

This script generates various plots with different themes to verify theming works correctly
"""

import json
import os
from pathlib import Path

import matplotlib.pyplot as plt
import numpy as np

# Source the theme function
import sys
sys.path.insert(0, str(Path(__file__).parent / "src"))
from theme import theme


def set_theme(variables):
    """Set a theme from Stencila plotting variables"""
    # Convert dict to JSON
    json_str = json.dumps(variables)
    # Apply theme
    theme(json_str)


def create_plots(output_dir):
    """Create various plots and save them to a folder"""
    # Create output directory
    Path(output_dir).mkdir(parents=True, exist_ok=True)

    # Get background color from rcParams
    bg = plt.rcParams.get("figure.facecolor", "white")

    # Generate sample data
    np.random.seed(42)
    x = np.arange(1, 51)
    y1 = np.cumsum(np.random.normal(0.5, 2, 50))
    y2 = np.cumsum(np.random.normal(-0.3, 1.5, 50))
    categories = ["A", "B", "C", "D", "E"]
    cat_values = [23, 45, 12, 67, 34]

    # --- MATPLOTLIB PLOTS ---

    # 1. Scatter plot
    fig, ax = plt.subplots(figsize=(9, 6))
    ax.scatter(x, y1, s=plt.rcParams["lines.markersize"]**2)
    ax.set_title("Matplotlib: Scatter Plot")
    ax.set_xlabel("X Axis")
    ax.set_ylabel("Y Axis")
    fig.savefig(os.path.join(output_dir, "01_matplotlib_scatter.png"))
    plt.close(fig)

    # 2. Line plot
    fig, ax = plt.subplots(figsize=(9, 6))
    ax.plot(x, y1, label="Series 1")
    ax.plot(x, y2, label="Series 2")
    ax.set_title("Matplotlib: Line Plot")
    ax.set_xlabel("X Axis")
    ax.set_ylabel("Y Axis")
    ax.set_ylim(-5, 30)
    ax.legend()
    fig.savefig(os.path.join(output_dir, "02_matplotlib_line.png"))
    plt.close(fig)

    # 3. Bar plot
    fig, ax = plt.subplots(figsize=(9, 6))
    bars = ax.bar(categories, cat_values)
    # Color each bar with sequential colors from the color cycle
    colors = [f"C{i}" for i in range(len(categories))]
    for bar, color in zip(bars, colors):
        bar.set_color(color)
    # Add legend matching ggplot2 behavior
    from matplotlib.patches import Patch
    legend_elements = [Patch(facecolor=f"C{i}", label=cat) for i, cat in enumerate(categories)]
    ax.legend(handles=legend_elements, title="category")
    ax.set_title("Matplotlib: Bar Plot")
    ax.set_xlabel("Category")
    ax.set_ylabel("Value")
    fig.savefig(os.path.join(output_dir, "03_matplotlib_bar.png"))
    plt.close(fig)

    # 4. Histogram
    fig, ax = plt.subplots(figsize=(9, 6))
    ax.hist(np.random.normal(0, 1, 1000), bins=30, color="C2")
    ax.set_title("Matplotlib: Histogram")
    ax.set_xlabel("Value")
    ax.set_ylabel("Frequency")
    fig.savefig(os.path.join(output_dir, "04_matplotlib_histogram.png"))
    plt.close(fig)

    # 5. Box plot
    fig, ax = plt.subplots(figsize=(9, 6))
    data_list = [
        np.random.normal(5, 1, 100),
        np.random.normal(7, 1, 100),
        np.random.normal(6, 1, 100),
        np.random.normal(8, 1, 100),
    ]
    bp = ax.boxplot(data_list, tick_labels=["A", "B", "C", "D"], patch_artist=True)
    # Color each box with sequential colors
    for patch, i in zip(bp["boxes"], range(len(data_list))):
        patch.set_facecolor(f"C{i}")
    ax.set_title("Matplotlib: Box Plot")
    ax.set_xlabel("Group")
    ax.set_ylabel("Value")
    fig.savefig(os.path.join(output_dir, "05_matplotlib_boxplot.png"))
    plt.close(fig)

    # 6. Multiple panels
    fig, ((ax1, ax2), (ax3, ax4)) = plt.subplots(2, 2, figsize=(9, 6))
    ax1.scatter(x, y1, s=plt.rcParams["lines.markersize"]**2, color="C0")
    ax1.set_title("Points")
    ax2.plot(x, y1, color="C1")
    ax2.set_title("Line")
    ax3.plot(x, y1, marker="o", color="C2")
    ax3.set_title("Both")
    bars = ax4.bar(categories[:4], cat_values[:4])
    for bar, i in zip(bars, range(4)):
        bar.set_color(f"C{i}")
    ax4.set_title("Bars")
    fig.tight_layout()
    fig.savefig(os.path.join(output_dir, "06_matplotlib_panels.png"))
    plt.close(fig)

    # 7. Annotated plot with subtitle and caption
    fig, ax = plt.subplots(figsize=(9, 6))
    ax.plot(x, y1, marker="o")
    ax.set_title("Annotated Plot")
    ax.set_xlabel("X Axis")
    ax.set_ylabel("Y Axis")
    # Use suptitle for subtitle effect (positioned below title)
    fig.suptitle("Matplotlib: Annotated Plot", y=0.98, fontsize=plt.rcParams["axes.titlesize"])
    ax.set_title("This is a subtitle to test subtitle styling", fontsize=plt.rcParams.get("axes.titlesize", 10) * 0.8, pad=10)
    # Add caption as text annotation at bottom
    fig.text(0.5, 0.02, "This is a caption to test caption styling",
             ha="center", fontsize=plt.rcParams["font.size"] * 0.9,
             color=plt.rcParams["text.color"])
    fig.savefig(os.path.join(output_dir, "07_matplotlib_annotated.png"), bbox_inches="tight")
    plt.close(fig)


# Three contrasting themes that we can compare to check that theme variables
# are being applied properly. No variable should be the same in all three themes.
themes = {
    # Monochrome: light background, grays, sans fonts
    # Uses axis-specific grid widths to demonstrate the feature
    "monochrome": {
        "plot-background": "#ffffff",
        "plot-padding-top": 4,
        "plot-padding-right": 4,
        "plot-padding-bottom": 8,
        "plot-padding-left": 8,
        "plot-axis-line-color": "#404040",
        "plot-axis-line-width": 0.5,
        "plot-axis-title-color": "#404040",
        "plot-axis-title-size": 11,
        "plot-color-1": "#2c2c2c",
        "plot-color-2": "#474747",
        "plot-color-3": "#5c5c5c",
        "plot-color-4": "#717171",
        "plot-color-5": "#858585",
        "plot-color-6": "#9a9a9a",
        "plot-color-7": "#adadad",
        "plot-color-8": "#c2c2c2",
        "plot-color-9": "#d6d6d6",
        "plot-color-10": "#e0e0e0",
        "plot-color-11": "#ebebeb",
        "plot-color-12": "#f0f0f0",
        "plot-font-family": "Liberation Sans",
        "plot-font-size": 10,
        "plot-grid-color": "#aaaaaa",
        "plot-grid-x-width": 0,  # No vertical bars
        "plot-grid-y-width": 1,  # Only horizontal
        "plot-legend-background": "#ffffff",
        "plot-legend-border-color": "#808080",
        "plot-legend-border-width": 1,
        "plot-legend-text-color": "#404040",
        "plot-legend-size": 9,
        "plot-legend-position": "right",
        "plot-line-cap": "round",
        "plot-line-join": "round",
        "plot-line-width": 1.5,
        "plot-panel-border-color": "#808080",
        "plot-panel-border-width": 1,
        "plot-point-size": 6,
        "plot-text-color": "#303030",
        "plot-tick-color": "#404040",
        "plot-tick-size": 4,
        "plot-tick-width": 1,
        "plot-title-size": 14,
        "plot-subtitle-size": 11,
    },
    # Vintage: parchment background, muted colors, serif fonts
    # Uses axis-specific grid widths with opposite emphasis from monochrome
    "vintage": {
        "plot-background": "#f1d8b8ff",
        "plot-padding-top": 12,
        "plot-padding-right": 12,
        "plot-padding-bottom": 24,
        "plot-padding-left": 24,
        "plot-axis-line-color": "#5a4a3a",
        "plot-axis-line-width": 1.5,
        "plot-axis-title-color": "#5a4a3a",
        "plot-axis-title-size": 12,
        "plot-color-1": "#8b4513",
        "plot-color-2": "#a0522d",
        "plot-color-3": "#cd853f",
        "plot-color-4": "#d2691e",
        "plot-color-5": "#bc8f8f",
        "plot-color-6": "#8b7355",
        "plot-color-7": "#6b8e23",
        "plot-color-8": "#808000",
        "plot-color-9": "#9acd32",
        "plot-color-10": "#b8860b",
        "plot-color-11": "#daa520",
        "plot-color-12": "#cd5c5c",
        "plot-font-family": "Liberation Serif",
        "plot-font-size": 11,
        "plot-grid-color": "#a75809ff",
        "plot-grid-x-width": 1.2,  # Thicker vertical grid lines
        "plot-grid-y-width": 0.4,  # Thinner horizontal grid lines
        "plot-legend-background": "#f8efe0",
        "plot-legend-border-color": "#8b7355",
        "plot-legend-border-width": 1.5,
        "plot-legend-text-color": "#5a4a3a",
        "plot-legend-size": 10,
        "plot-legend-position": "bottom",
        "plot-line-cap": "round",
        "plot-line-join": "round",
        "plot-line-width": 2,
        "plot-panel-border-color": "#8b7355",
        "plot-panel-border-width": 1.5,
        "plot-point-size": 8,
        "plot-text-color": "#4a3a2a",
        "plot-tick-color": "#5a4a3a",
        "plot-tick-size": 8,
        "plot-tick-width": 1.5,
        "plot-title-size": 16,
        "plot-subtitle-size": 12,
    },
    # Cyberpunk: dark background, bright colors, monospace fonts
    "cyberpunk": {
        "plot-background": "#0a0e27",
        "plot-padding-top": 20,
        "plot-padding-right": 20,
        "plot-padding-bottom": 40,
        "plot-padding-left": 40,
        "plot-axis-line-color": "#00ffff",
        "plot-axis-line-width": 2,
        "plot-axis-title-color": "#00ffff",
        "plot-axis-title-size": 10,
        "plot-color-1": "#00ffff",
        "plot-color-2": "#ff00ff",
        "plot-color-3": "#ff0080",
        "plot-color-4": "#00ff00",
        "plot-color-5": "#0080ff",
        "plot-color-6": "#ffff00",
        "plot-color-7": "#ff8000",
        "plot-color-8": "#8000ff",
        "plot-color-9": "#ff0040",
        "plot-color-10": "#00ff80",
        "plot-color-11": "#80ff00",
        "plot-color-12": "#ff00c0",
        "plot-font-family": "Liberation Mono",
        "plot-font-size": 12,
        "plot-grid-color": "#178a1bff",
        "plot-grid-width": 1,
        "plot-legend-background": "#151933",
        "plot-legend-border-color": "#ff00ff",
        "plot-legend-border-width": 2,
        "plot-legend-text-color": "#00ffff",
        "plot-legend-size": 8,
        "plot-legend-position": "top",
        "plot-line-cap": "butt",
        "plot-line-join": "bevel",
        "plot-line-width": 2.5,
        "plot-panel-border-color": "#ff00ff",
        "plot-panel-border-width": 2,
        "plot-point-size": 10,
        "plot-text-color": "#00ffff",
        "plot-tick-color": "#00ffff",
        "plot-tick-size": 3,
        "plot-tick-width": 2,
        "plot-title-size": 18,
        "plot-subtitle-size": 16,
    },
}


def run_tests():
    """Run tests for each theme"""
    for theme_name in themes:
        print(f"Generating plots for theme: {theme_name}")
        set_theme(themes[theme_name])
        create_plots(os.path.join("test-themes", theme_name))


# Run the tests
if __name__ == "__main__":
    run_tests()

File: rust/kernel-r/test-themes.r
#!/usr/bin/env Rscript

# Test script for visual verification of Stencila theme variables
# This script generates various plots with different themes to verify theming works correctly

library(jsonlite)

# Source the theme_ function
source("src/theme.r")

# Function to set a theme from Stencila plotting variables
set_theme <- function(variables) {
  # Convert R list to JSON
  json <- toJSON(variables, auto_unbox = TRUE)

  # Apply theme
  theme_(json)
}

# Function to create various plots and save them to a folder
create_plots <- function(output_dir) {
  # Create output directory
  dir.create(output_dir, recursive = TRUE, showWarnings = FALSE)

  # Get background color and font size from theme options
  bg <- getOption("stencila.plot.background", "white")
  ps <- getOption("stencila.plot.font.size", 12)

  # Generate sample data
  set.seed(42)
  x <- 1:50
  y1 <- cumsum(rnorm(50, mean = 0.5, sd = 2))
  y2 <- cumsum(rnorm(50, mean = -0.3, sd = 1.5))
  categories <- c("A", "B", "C", "D", "E")
  cat_values <- c(23, 45, 12, 67, 34)

  # --- BASE R PLOTS ---

  # 1. Scatter plot
  png_file <- file.path(output_dir, "01_base_scatter.png")
  png(png_file, width = 9, height = 6, units = "in", res = 100, bg = bg)
  plot(x, y1,
       main = "Base R: Scatter Plot",
       xlab = "X Axis",
       ylab = "Y Axis",
       pch = 19)
  dev.off()
  
  # 2. Line plot
  png_file <- file.path(output_dir, "02_base_line.png")
  png(png_file, width = 9, height = 6, units = "in", res = 100, bg = bg)
  plot(x, y1, type = "l",
       main = "Base R: Line Plot",
       ylim = c(-5, 30),
       xlab = "X Axis",
       ylab = "Y Axis",
       col = 1)
  lines(x, y2, col = 2)
  legend("topright", legend = c("Series 1", "Series 2"),
         col = 1:2, lty = 1)
  dev.off()

  # 3. Bar plot
  png_file <- file.path(output_dir, "03_base_bar.png")
  png(png_file, width = 9, height = 6, units = "in", res = 100, bg = bg)
  barplot(cat_values,
          names.arg = categories,
          main = "Base R: Bar Plot",
          xlab = "Category",
          ylab = "Value",
          col = 1:5)
  dev.off()

  # 4. Histogram
  png_file <- file.path(output_dir, "04_base_histogram.png")
  png(png_file, width = 9, height = 6, units = "in", res = 100, bg = bg)
  hist(rnorm(1000),
       main = "Base R: Histogram",
       xlab = "Value",
       ylab = "Frequency",
       col = 3,
       breaks = 30)
  dev.off()

  # 5. Box plot
  png_file <- file.path(output_dir, "05_base_boxplot.png")
  png(png_file, width = 9, height = 6, units = "in", res = 100, bg = bg)
  data_list <- list(
    A = rnorm(100, mean = 5),
    B = rnorm(100, mean = 7),
    C = rnorm(100, mean = 6),
    D = rnorm(100, mean = 8)
  )
  boxplot(data_list,
          main = "Base R: Box Plot",
          xlab = "Group",
          ylab = "Value",
          col = 1:4)
  dev.off()

  # 6. Multiple panels
  png_file <- file.path(output_dir, "06_base_panels.png")
  png(png_file, width = 9, height = 6, units = "in", res = 100, bg = bg, pointsize = ps)
  par(mfrow = c(2, 2))
  plot(x, y1, type = "p", main = "Points", col = 1, pch = 19)
  plot(x, y1, type = "l", main = "Line", col = 2)
  plot(x, y1, type = "b", main = "Both", col = 3, pch = 19)
  barplot(cat_values[1:4], main = "Bars", col = 1:4)
  par(mfrow = c(1, 1))
  dev.off()

  # --- GGPLOT2 PLOTS (if available) ---

  if (requireNamespace("ggplot2", quietly = TRUE)) {
    library(ggplot2)

    # Create data frames for ggplot2
    df1 <- data.frame(x = x, y1 = y1, y2 = y2)
    df2 <- data.frame(category = categories, value = cat_values)
    df3 <- data.frame(value = rnorm(1000))
    df4 <- data.frame(
      group = rep(c("A", "B", "C", "D"), each = 100),
      value = c(rnorm(100, 5), rnorm(100, 7), rnorm(100, 6), rnorm(100, 8))
    )

    # 7. ggplot2 Scatter
    png_file <- file.path(output_dir, "07_ggplot_scatter.png")
    p <- ggplot(df1, aes(x = x, y = y1)) +
      geom_point() +
      labs(title = "ggplot2: Scatter Plot",
           x = "X Axis",
           y = "Y Axis")
    ggsave(png_file, p, width = 9, height = 6, dpi = 100)

    # 8. ggplot2 Line
    png_file <- file.path(output_dir, "08_ggplot_line.png")
    df_long <- data.frame(
      x = rep(x, 2),
      y = c(y1, y2),
      series = rep(c("Series 1", "Series 2"), each = length(x))
    )
    p <- ggplot(df_long, aes(x = x, y = y, color = series)) +
      geom_line() +
      labs(title = "ggplot2: Line Plot",
           x = "X Axis",
           y = "Y Axis",
           color = "Series")
    ggsave(png_file, p, width = 9, height = 6, dpi = 100)

    # 9. ggplot2 Bar
    png_file <- file.path(output_dir, "09_ggplot_bar.png")
    p <- ggplot(df2, aes(x = category, y = value, fill = category)) +
      geom_bar(stat = "identity") +
      labs(title = "ggplot2: Bar Plot",
           x = "Category",
           y = "Value")
    ggsave(png_file, p, width = 9, height = 6, dpi = 100)

    # 10. ggplot2 Histogram
    png_file <- file.path(output_dir, "10_ggplot_histogram.png")
    p <- ggplot(df3, aes(x = value)) +
      geom_histogram(bins = 30) +
      labs(title = "ggplot2: Histogram",
           x = "Value",
           y = "Count")
    ggsave(png_file, p, width = 9, height = 6, dpi = 100)

    # 11. ggplot2 Boxplot
    png_file <- file.path(output_dir, "11_ggplot_boxplot.png")
    p <- ggplot(df4, aes(x = group, y = value, fill = group)) +
      geom_boxplot() +
      labs(title = "ggplot2: Box Plot",
           x = "Group",
           y = "Value")
    ggsave(png_file, p, width = 9, height = 6, dpi = 100)

    # 12. ggplot2 Faceted plot
    png_file <- file.path(output_dir, "12_ggplot_facet.png")
    p <- ggplot(df4, aes(x = value)) +
      geom_histogram(bins = 20) +
      facet_wrap(~ group, ncol = 2) +
      labs(title = "ggplot2: Faceted Histogram",
           x = "Value",
           y = "Count")
    ggsave(png_file, p, width = 9, height = 6, dpi = 100)

    # 13. ggplot2 with subtitle and caption
    png_file <- file.path(output_dir, "13_ggplot_annotated.png")
    p <- ggplot(df1, aes(x = x, y = y1)) +
      geom_line() +
      geom_point() +
      labs(title = "ggplot2: Annotated Plot",
           subtitle = "This is a subtitle to test subtitle styling",
           caption = "This is a caption to test caption styling",
           x = "X Axis",
           y = "Y Axis")
    ggsave(png_file, p, width = 9, height = 6, dpi = 100)
  } else {
    cat("ggplot2 not available, skipping ggplot2 plots\n")
  }
}

# Three contrasting themes that we can compare to check that theme variables
# are being applied properly. No variable should be the same in all three themes.
themes <- list(
  # Monochrome: light background, grays, sans fonts
  # Uses axis-specific grid widths to demonstrate the feature
  "monochrome" = list(
    "plot-background" = "#ffffff",
    "plot-panel" = "#f5f5f5",
    "plot-padding-top" = 2,
    "plot-padding-right" = 2,
    "plot-padding-bottom" = 2,
    "plot-padding-left" = 2,
    "plot-axis-line-color" = "#404040",
    "plot-axis-line-width" = 0.5,
    "plot-axis-title-color" = "#404040",
    "plot-axis-title-size" = 11,
    "plot-color-1" = "#2c2c2c",
    "plot-color-2" = "#474747",
    "plot-color-3" = "#5c5c5c",
    "plot-color-4" = "#717171",
    "plot-color-5" = "#858585",
    "plot-color-6" = "#9a9a9a",
    "plot-color-7" = "#adadad",
    "plot-color-8" = "#c2c2c2",
    "plot-color-9" = "#d6d6d6",
    "plot-color-10" = "#e0e0e0",
    "plot-color-11" = "#ebebeb",
    "plot-color-12" = "#f0f0f0",
    "plot-font-family" = "Liberation Sans",
    "plot-font-size" = 10,
    "plot-grid-color" = "#aaaaaa",
    "plot-grid-x-width" = 0,  # No vertical bars
    "plot-grid-y-width" = 1,  # Only horizontal
    "plot-legend-background" = "#ffffff",
    "plot-legend-border-color" = "#808080",
    "plot-legend-border-width" = 1,
    "plot-legend-text-color" = "#404040",
    "plot-legend-position" = "right",
    "plot-line-width" = 1.5,
    "plot-point-size" = 6,
    "plot-text-color" = "#303030",
    "plot-title-size" = 14,
    "plot-subtitle-size" = 11
  ),

  # Vintage: parchment background, muted colors, serif fonts
  # Uses axis-specific grid widths with opposite emphasis from monochrome
  "vintage" = list(
    "plot-background" = "#f1d8b8ff",
    "plot-panel" = "#f8efe0",
    "plot-padding-top" = 10,
    "plot-padding-right" = 20,
    "plot-padding-bottom" = 30,
    "plot-padding-left" = 40,
    "plot-axis-line-color" = "#5a4a3a",
    "plot-axis-line-width" = 1.5,
    "plot-axis-title-color" = "#5a4a3a",
    "plot-axis-title-size" = 12,
    "plot-color-1" = "#8b4513",
    "plot-color-2" = "#a0522d",
    "plot-color-3" = "#cd853f",
    "plot-color-4" = "#d2691e",
    "plot-color-5" = "#bc8f8f",
    "plot-color-6" = "#8b7355",
    "plot-color-7" = "#6b8e23",
    "plot-color-8" = "#808000",
    "plot-color-9" = "#9acd32",
    "plot-color-10" = "#b8860b",
    "plot-color-11" = "#daa520",
    "plot-color-12" = "#cd5c5c",
    "plot-font-family" = "Liberation Serif",
    "plot-font-size" = 11,
    "plot-grid-color" = "#a75809ff",
    "plot-grid-x-width" = 1.2,  # Thicker vertical grid lines
    "plot-grid-y-width" = 0.4,  # Thinner horizontal grid lines
    "plot-legend-background" = "#f8efe0",
    "plot-legend-border-color" = "#8b7355",
    "plot-legend-border-width" = 1.5,
    "plot-legend-text-color" = "#5a4a3a",
    "plot-legend-position" = "bottom",
    "plot-line-width" = 2,
    "plot-point-size" = 8,
    "plot-text-color" = "#4a3a2a",
    "plot-title-size" = 16,
    "plot-subtitle-size" = 12
  ),

  # Cyberpunk: dark background, bright colors, monospace fonts
  "cyberpunk" = list(
    "plot-background" = "#0a0e27",
    "plot-panel" = "#151933",
    "plot-padding-top" = 20,
    "plot-padding-right" = 20,
    "plot-padding-bottom" = 40,
    "plot-padding-left" = 40,
    "plot-axis-line-color" = "#00ffff",
    "plot-axis-line-width" = 2,
    "plot-axis-title-color" = "#00ffff",
    "plot-axis-title-size" = 10,
    "plot-color-1" = "#00ffff",
    "plot-color-2" = "#ff00ff",
    "plot-color-3" = "#ff0080",
    "plot-color-4" = "#00ff00",
    "plot-color-5" = "#0080ff",
    "plot-color-6" = "#ffff00",
    "plot-color-7" = "#ff8000",
    "plot-color-8" = "#8000ff",
    "plot-color-9" = "#ff0040",
    "plot-color-10" = "#00ff80",
    "plot-color-11" = "#80ff00",
    "plot-color-12" = "#ff00c0",
    "plot-font-family" = "Liberation Mono",
    "plot-font-size" = 12,
    "plot-grid-color" = "#178a1bff",
    "plot-grid-width" = 1,
    "plot-legend-background" = "#151933",
    "plot-legend-border-color" = "#ff00ff",
    "plot-legend-border-width" = 2,
    "plot-legend-text-color" = "#00ffff",
    "plot-legend-position" = "top",
    "plot-line-width" = 2.5,
    "plot-point-size" = 10,
    "plot-text-color" = "#00ffff",
    "plot-title-size" = 18,
    "plot-subtitle-size" = 16
  )
)

# Run tests for each theme
run_tests <- function() {
  for (theme in names(themes)) {
    cat("Generating plots for theme: ", theme, "\n")
    set_theme(themes[[theme]])
    create_plots(file.path("test-themes", theme))
  }
}

# Run the tests
if (!interactive()) {
  run_tests()
}
